src/summary_table/SummaryTable.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useMemo } from 'react';
+import Box from '@mui/material/Box';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableRow from '@mui/material/TableRow';
+
+import Paper from '@mui/material/Paper';
+
+import { SortableTableHeader } from './SortableTableHeader';
+import { headerCells, Data, rows } from './data';
+
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+type Order = 'asc' | 'desc';
+
+function getComparator<Key extends keyof any>(order: Order, orderBy: Key): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+  return order === 'desc' ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+const SummaryTable = (props: any) => {
+  const [order, setOrder] = useState<Order>('desc');
+  const [orderBy, setOrderBy] = useState<keyof Data>('type');
+
+  const handleRequestSort = (property: keyof Data) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const sortedRows = useMemo(() => rows.sort(getComparator(order, orderBy)), [order, orderBy]);
+  const alignment = (value: number | string) => (typeof value === 'number' ? 'right' : 'left');
+  return (
+    <Box sx={{ width: '100%' }}>
```

Comment:
This container is not doing anything. Since `<SummaryTable>` may be put into a container by it's parent, probably this `Box` is not needed.